packages/design-system/src/TreeDropdown/index.tsx

```diff
@@ -267,6 +266,72 @@ function getSelectedOption(
   return selectedOption;
 }
 
+type RenderTreeOptionProps = {
+  option: TreeDropdownOption;
+  optionTree: TreeDropdownOption[];
+  selectedOption: TreeDropdownOption;
+  handleSelect: (
+    option: TreeDropdownOption,
+    isUpdatedViaKeyboard: boolean,
+  ) => any;
+  handleOptionClick: (
+    option: TreeDropdownOption,
+  ) => (e: any, isUpdatedViaKeyboard?: boolean) => void;
+};
+
+function RenderTreeOption({
+  handleOptionClick,
+  handleSelect,
+  option,
+  optionTree,
```

Comment:
This function has been moved out of the TreeDropdown component, because earlier we were getting the following error while updating the options.
![Screenshot 2023-01-04 at 1 06 17 PM](https://user-images.githubusercontent.com/13567359/210564199-ff2c04bc-501a-4431-b2e8-882200a1344d.png)

We're using `useMemo` and `useCallback` hooks inside this component. And upon changing the options, the number of hooks being called was changing in between renders.
By extracting this component out, we don't run into such an issue.